src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java

```diff
@@ -641,24 +641,23 @@ private void findMatch() {
                     logMessage(startLine);
                 }
             }
-            if (canContinueValidation(ignore)) {
-                findMatch();
+            if (!canContinueValidation()) {
+                cont = false;
```

Comment:
can it be `shouldContinue = canContinueValidation()` ?

Reply:
fixed. I removed shouldContinue completely. It still passes all test cases